node/hyperbahn/service_proxy.js

```diff
@@ -257,6 +262,64 @@ function updateExitNodes(exitNodes, svcchan) {
     }
 };
 
+ServiceDispatchHandler.prototype.isBlocked =
+function isBlocked(cn, serviceName) {
+    var self = this;
+    if (!self.blockingTable) {
+        return false;
+    }
+
+    cn = cn || '*';
+    serviceName = serviceName || '*';
+
+    if (self.blockingTable[cn + '~~' + serviceName] ||
+        self.blockingTable['*~~' + serviceName] ||
+        self.blockingTable[cn + '~~*']) {
+        return true;
+    }
+
+    return false;
+};
+
+ServiceDispatchHandler.prototype.block =
+function block(cn, serviceName) {
+    var self = this;
+    cn = cn || '*';
+    serviceName = serviceName || '*';
+    self.blockingTable = self.blockingTable || {};
+    assert(cn !== '*' || serviceName !== '*', 'at least one of cn/serviceName should be provided');
+    var key = cn + '~~' + serviceName;
+    self.blockingTable[key] = Date.now();
+};
+
+ServiceDispatchHandler.prototype.unblock =
+function unblock(cn, serviceName) {
+    var self = this;
+    if (!self.blockingTable) {
+        return;
+    }
+
+    cn = cn || '*';
+    serviceName = serviceName || '*';
+    
+    var keys = Object.keys(self.blockingTable);
```

Comment:
dont do this complicated clearing stuff. Just do `delete blockingTable[cn + ~~ + serviceName]`